zschema/compounds.py

```diff
@@ -159,7 +155,9 @@ def merge(self, other):
                 newdef[key] = l_value.merge(r_value)
             else:
                 raise MergeConflictException("Only subrecords can be merged. (%s)", key)
-        self.definition = newdef
+        self.set("definition", newdef)
+        self.set("required", self.required or other.required)
+        self.set("doc", self.doc or other.doc)
```

Comment:
I wouldn't have expected `required` / `doc` to get merged, but since that was at least the previous intent (if not the actual behavior), that would be an argument in favor of dealing with that later.